.github/workflows/coveo-itertools.yml

```diff
@@ -0,0 +1,52 @@
+name: coveo-itertools
+on:
+  push:
+    paths:
+      - 'coveo-itertools/**'
+  workflow_dispatch: {}
+
+jobs:
+  pyprojectci:
+    name: pyproject ci
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.8, 3.9]
+        os: [ubuntu-20.04, windows-2019, macos-10.15]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Run pyproject ci
+        uses: ./.github/workflows/actions/pyproject-ci
+        with:
+          project-name: ${{ github.workflow }}
+
+  publish:
+    name: Publish to pypi.org
+    runs-on: ubuntu-20.04
+    needs: pyprojectci
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Publish to pypi
+        uses: ./.github/workflows/actions/publish-to-pypi
+        with:
+          project-name: ${{ github.workflow }}
+          pypi-token: ${{ secrets.PYPI_TOKEN }}
+          dry-run: ${{ github.ref != 'refs/heads/main' }}

```

Comment:
I'm starting to guess that I'll review the same workflow with different `name` and `paths`. Perhaps you can write a local composite-action with parameterized `name`? I think you'd reap the benefit after you decide to update the workflows 2-3 times (for instance to update `python-version`).